File: Algorithms/Implementation/Electronics Shop.js
// Electronics Shop

// https://www.hackerrank.com/challenges/electronics-shop/problem


// #1
function getMoneySpent(keyboards, drives, b) {
  keyboards.sort((a, b) => a - b)
  drives.sort((a, b) => a - b)
  let maxCost = -1
  let k = keyboards.length - 1
  while (k >= 0) {
    if (keyboards[k] >= b) {
      k--
      continue;
    }
    let d = drives.length - 1
    let currCost
    while (d >= 0) {
      currCost = keyboards[k] + drives[d]
      if (currCost > b) {
        d--
        continue;
      }
      if (currCost <= b) {
        maxCost = Math.max(maxCost, currCost)
        break;
      }
      d--
    }
    k--
  }
  return maxCost
}

File: Algorithms/Implementation/Electronics Shop.py
# Electronics Shop

def getMoneySpent(keyboards, drives, b):
  maxCost = -1
  K = sorted(keyboards, reverse = True)
  D = sorted(drives, reverse = True)
    
  for i in range(len(K)):
    currMax = K[i]

    if currMax >= b:
      continue
            
    for j in range(len(D)):
      currMax += D[j]
            
      if currMax > b:
        currMax -= D[j]
        continue
            
      maxCost = max(maxCost, currMax)
      break
    
  return maxCost